### DISCUSSION TITLE
Problem with low frequency motor vibration at GEPRC_F722_AIO

### DISCUSSION THREAD

**Discussion by kohones** [iNavFlight/inav#9054] [Q&A]
Hi, recently I've tried to migrate from betaflight 4.4.1 to INAV. I managaed to get through configuration without bigger issues. Drone is flyable, but there is noticeable wobble especially when using more throttle. I did some analysis in blackbox explorer and I noticed some vibration coming from motors at 40-50Hz. On betaflight there is no such vibration - drone is flying perfectly smooth. 

Firstly I thought that there can be some problem with PID tuning. But i've tried to tune regulator and I was checking many combinations of PID parameters but vibration still exists. Only time when vibration were noticeably smaller was when I've set very low PID terms (eg. P:20, I:20, D:20). Problem exists on INAV version 6.0 and also 6.1. I've check mechanics and there is no loose parts or play on motor bearings. 

Can someone give me some advice what can be wrong with configuration? What should I check?
Is someone succesfully flying with INAV on this drone? 

Blackbox log:
[blackbox_log_2023-05-17_180908_STOCK.TXT](https://github.com/iNavFlight/inav/files/11519601/blackbox_log_2023-05-17_180908_STOCK.TXT)

INAV configuration 
[P1INAV_6.1.0_cli_20230516_162941.txt](https://github.com/iNavFlight/inav/files/11519609/P1INAV_6.1.0_cli_20230516_162941.txt)

![scr](https://github.com/iNavFlight/inav/assets/24231253/b555eb3d-2638-4fc6-8172-9a8aa789620e)

Configuration:
GEPRC Cinebot 30 - 6S
FC: GEP-F722-45A AIO V2
MCU: STM32F722RET6
IMU: ICM-42688-P
VTX: Walksnail
Props: HQ prop T76mm
Motors: SPEEDX2 1804-2450KV

**Comment by Jetrell** [comment]
Can you post a DIFF ALL config file. Its easier to separate what you've changed, than a DUMP file.
Were you using the RPM filter in Betaflight ? Can you provide a Raw gyro spectrograph of the BF log... I would be interested to see what it looks like.. Mostly because 40 - 50Hz noise is in the IMU's usable data range, that the FC uses for controlling the quad.. RPM filtering maybe able to eliminate some of this noise. But not remove it entirely... Remember that BF has Bi-directional Dshot for RPM filtering. While INAV does not.

Are your tests done in Acro or Angle mode.. Or in a Navigation mode ?

**Reply by kohones** [reply]
I'm flying on Acro mode. 

Here is 'diff' (currently i dont have any 'dif all') file from INAV:
[P1INAV_6.0.0_cli_20230513_214910.txt](https://github.com/iNavFlight/inav/files/11521678/P1INAV_6.0.0_cli_20230513_214910.txt)

Here are some graphs from betaflight and also blackbox recording (btw. this is probably not gyro raw data).

![btfl1](https://github.com/iNavFlight/inav/assets/24231253/ccb32bd4-f449-4d29-b0bf-1e9f7decd0b3)
![btfl2](https://github.com/iNavFlight/inav/assets/24231253/88333716-36a9-4ab7-8a6a-b7fc97fb1c03)


[BTFL_BLACKBOX_LOG_Cinebot30_20230421_002831.zip](https://github.com/iNavFlight/inav/files/11521697/BTFL_BLACKBOX_LOG_Cinebot30_20230421_002831.zip)

I did some tests while holding drone mid air and flat in hand (yeah I'm aware of the risk ;) ). I thought that it can be good method to eliminate some external variables (like wind for example). So I was holding drone and I was raising throttle to the 50%. Vibrations were present (you can feel it, and also you can hear it) on INAV. But when I flashed BF it was completely gone. During this test i also tried to disable RPM filtering on BF and this not change anything (everything was perfectly smooth). 

Currently I'm on BF so I can get more data from it if needed. If you have some test scenarios, I can also back to INAV to check.

**Reply by Jetrell** [reply]
The first thing I would try, is make sure  `set i2c_speed = 800KHZ` for the magnetometer to work at its best.
Then increase the looptime... `set looptime = 2000`.
INAV runs a slower looptime than BF, due to it being more orientated towards sensory navigation. 
Running it higher (Not more than 2Khz) will allow for better performance and  can reduce the vibration levels in _some circumstances_. Because it increases the motor control update rate. Which also means, noise spikes will have less time to rise in amplitude.

**Reply by kohones** [reply]
I2C clock speed is set to 800k. Both sensors are working. I've checked the loop time on BF and 8k, 2k and 1k doesn't produce any wobbling. When I flash inav i will check different loop times there.
